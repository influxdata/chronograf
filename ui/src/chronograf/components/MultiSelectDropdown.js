import React, {PropTypes} from 'react';
import OnClickOutside from 'shared/components/OnClickOutside';
import classNames from 'classnames';
import _ from 'lodash';

const {func, arrayOf, string} = PropTypes;
const MultiSelectDropdown = React.createClass({
  propTypes: {
    onApply: func.isRequired,
    items: arrayOf(PropTypes.string.isRequired).isRequired,
    selectedItems: arrayOf(string.isRequired).isRequired,
  },

  getInitialState() {
    return {
      isOpen: false,
      localSelectedItems: this.props.selectedItems,
    };
  },

  componentWillReceiveProps(nextProps) {
    if (!_.isEqual(this.state.localSelectedItems, nextProps.selectedItems)) {
      this.setState({
        localSelectedItems: nextProps.selectedItems,
      });
    }
  },

  handleClickOutside() {
    this.setState({isOpen: false});
  },

  toggleMenu(e) {
    e.stopPropagation();
    this.setState({isOpen: !this.state.isOpen});
  },

  onSelect(item, e) {
    e.stopPropagation();

    const {localSelectedItems} = this.state;

    let nextItems;
    if (this.isSelected(item)) {
      nextItems = localSelectedItems.filter((i) => i !== item);
    } else {
      nextItems = localSelectedItems.concat(item);
    }

    this.setState({localSelectedItems: nextItems});
  },

  isSelected(item) {
    return this.state.localSelectedItems.indexOf(item) > -1;
  },

  onApplyFunctions(e) {
    e.stopPropagation();

    this.setState({isOpen: false});
    this.props.onApply(this.state.localSelectedItems);
  },

  render() {
    const {localSelectedItems} = this.state;

    return (
      <div className="dropdown multi-select-dropdown">
        <div onClick={this.toggleMenu} className="btn btn-xs btn-info dropdown-toggle multi-select-dropdown__toggle" type="button">
          <span className="multi-select-dropdown__label">
            {
              localSelectedItems.length ? localSelectedItems.map((s) => s).join(', ') : 'Apply Function'
            }
          </span>
          <span className="caret multi-select-dropdown__caret"></span>
        </div>
        {this.renderMenu()}
      </div>
    );
  },

  renderMenu() {
    const {items} = this.props;
    const {isOpen} = this.state;

    if (!isOpen) {
      return null;
    }

    return (
      <div className="dropdown-options">
        <ul className="dropdown-menu multi-select-dropdown__menu" aria-labelledby="dropdownMenu1">
          {items.map((listItem, i) => {
            return (
              <li className={classNames('multi-select-dropdown__item', {active: this.isSelected(listItem)})} key={i} onClick={_.wrap(listItem, this.onSelect)}>
                <a href="#">{listItem}</a>
              </li>
            );
          })}
          <li className="multi-select-dropdown__apply" onClick={this.onApplyFunctions}>
            <div className="btn btn-sm btn-primary btn-block">Apply</div>
          </li>
        </ul>
      </div>
    );
  },
});

export default OnClickOutside(MultiSelectDropdown);
