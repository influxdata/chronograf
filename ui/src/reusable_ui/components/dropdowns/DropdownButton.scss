/*
  Dropdown Button
  ------------------------------------------------------------------------------
*/

/* Button */
.dropdown--button {
  position: relative;
}

.dropdown--selected,
.dropdown--icon,
.dropdown--caret {
  position: absolute;
  top: 50%;
  transform: translateY(-50%);
}

.dropdown--icon {
  top: 48%;
}

.dropdown--selected {
  text-align: left;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

.dropdown--caret {
  margin: 0;
  font-size: 0.9em;
}

/* Button Size Modifiers */
@mixin buttonSizing($padding, $font) {
  .dropdown--icon,
  .dropdown--selected {
    left: $padding;
  }
  .dropdown--selected {
    width: calc(100% - #{($padding * 2) + $font});
  }
  .dropdown--icon + .dropdown--selected {
    left: $padding + ($font * 1.5);
    width: calc(100% - #{($padding * 2) + ($font * 2.5)});
  }
  .dropdown--caret {
    right: $padding;
  }
}

.dropdown--button.button-xs {
  @include buttonSizing($form-xs-padding, $form-xs-font);
}

.dropdown--button.button-sm {
  @include buttonSizing($form-sm-padding, $form-sm-font);
}

.dropdown--button.button-md {
  @include buttonSizing($form-md-padding, $form-md-font);
}

.dropdown--button.button-lg {
  @include buttonSizing($form-lg-padding, $form-lg-font);
}
