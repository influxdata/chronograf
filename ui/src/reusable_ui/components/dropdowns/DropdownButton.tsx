// Libraries
import React, {Component} from 'react'
import classnames from 'classnames'

// Types
import {
  ComponentStatus,
  ComponentColor,
  ComponentSize,
  IconFont,
  DropdownChild,
} from 'src/reusable_ui/types'

import {ErrorHandling} from 'src/shared/decorators/errors'

interface Props {
  children: DropdownChild
  onClick: () => void
  status?: ComponentStatus
  active?: boolean
  color?: ComponentColor
  size?: ComponentSize
  icon?: IconFont
}

@ErrorHandling
class DropdownButton extends Component<Props> {
  public static defaultProps: Partial<Props> = {
    color: ComponentColor.Default,
    size: ComponentSize.Small,
    status: ComponentStatus.Default,
    active: false,
  }

  public render() {
    const {onClick, status, children} = this.props
    return (
      <button
        className={this.classname}
        onClick={onClick}
        disabled={status === ComponentStatus.Disabled}
      >
        {this.icon}
        <span className="dropdown--selected">{children}</span>
        <span className="dropdown--caret icon caret-down" />
      </button>
    )
  }

  private get classname(): string {
    const {status, active, color, size} = this.props

    return classnames('dropdown--button button', {
      'button-stretch': true,
      [`button-${color}`]: color,
      [`button-${size}`]: size,
      disabled: status === ComponentStatus.Disabled,
      active,
    })
  }

  private get icon(): JSX.Element {
    const {icon} = this.props

    if (icon) {
      return <span className={`dropdown--icon icon ${icon}`} />
    }

    return null
  }
}

export default DropdownButton
