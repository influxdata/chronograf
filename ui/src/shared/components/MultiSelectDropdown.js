import React, {Component, PropTypes} from 'react'
import OnClickOutside from 'shared/components/OnClickOutside'
import classnames from 'classnames'
import _ from 'lodash'

const labelText = ({localSelectedItems, isOpen, label}) => {
  if (label) {
    return label
  } else if (localSelectedItems.length) {
    return localSelectedItems.map(s => s).join(', ')
  }

  // TODO: be smarter about the text displayed here
  if (isOpen) {
    return '0 Selected'
  }
  return 'None'
}

class MultiSelectDropdown extends Component {
  constructor(props) {
    super(props)

    this.state = {
      isOpen: false,
      localSelectedItems: this.props.selectedItems,
    }

    this.onSelect = ::this.onSelect
    this.onApplyFunctions = ::this.onApplyFunctions
  }

  handleClickOutside() {
    this.setState({isOpen: false})
  }

  toggleMenu(e) {
    e.stopPropagation()
    this.setState({isOpen: !this.state.isOpen})
  }

  onSelect(item, e) {
    e.stopPropagation()

    const {localSelectedItems} = this.state

    let nextItems
    if (this.isSelected(item)) {
      nextItems = localSelectedItems.filter(i => i !== item)
    } else {
      nextItems = [...localSelectedItems, item]
    }

    this.setState({localSelectedItems: nextItems})
  }

  isSelected(item) {
    return !!this.state.localSelectedItems.find(text => text === item)
  }

  onApplyFunctions(e) {
    e.stopPropagation()

    this.setState({isOpen: false})
    this.props.onApply(this.state.localSelectedItems)
  }

  render() {
    const {localSelectedItems, isOpen} = this.state
    const {label} = this.props

    return (
      <div
        className={classnames('dropdown multi-select-dropdown', {open: isOpen})}
      >
        <div
          onClick={::this.toggleMenu}
          className="btn btn-xs btn-info dropdown-toggle"
          type="button"
        >
          <div className="multi-select-dropdown__label">
            {labelText({localSelectedItems, isOpen, label})}
          </div>
          <span className="caret" />
        </div>
        {this.renderMenu()}
      </div>
    )
  }

  renderMenu() {
    const {items} = this.props

    return (
      <div className="dropdown-options">
        <div
          className="multi-select-dropdown__apply"
          onClick={this.onApplyFunctions}
          style={{listStyle: 'none'}}
        >
          <div className="btn btn-xs btn-info btn-block">Apply</div>
        </div>
        <ul
          className="dropdown-menu multi-select-dropdown__menu"
          aria-labelledby="dropdownMenu1"
        >
          {items.map((listItem, i) => {
            return (
              <li
                key={i}
                className={classnames('multi-select-dropdown__item', {
                  active: this.isSelected(listItem),
                })}
                onClick={_.wrap(listItem, this.onSelect)}
              >
                <a href="#">{listItem}</a>
              </li>
            )
          })}
        </ul>
      </div>
    )
  }
}

const {arrayOf, func, string} = PropTypes

MultiSelectDropdown.propTypes = {
  onApply: func.isRequired,
  items: arrayOf(string.isRequired).isRequired,
  selectedItems: arrayOf(string.isRequired).isRequired,
  label: string,
}

export default OnClickOutside(MultiSelectDropdown)
