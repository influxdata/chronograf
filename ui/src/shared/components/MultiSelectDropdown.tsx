import * as React from 'react'
import * as PropTypes from 'prop-types'

import classnames from 'classnames'
import * as _ from 'lodash'

import onClickOutside from 'shared/components/onClickOutside'
import FancyScrollbar from 'shared/components/FancyScrollbar'
import {DROPDOWN_MENU_MAX_HEIGHT} from 'shared/constants/index'

const labelText = ({localSelectedItems, isOpen, label}) => {
  if (localSelectedItems.length) {
    return localSelectedItems.map(s => s.name).join(', ')
  }

  if (label) {
    return label
  }

  // TODO: be smarter about the text displayed here
  if (isOpen) {
    return '0 Selected'
  }

  return 'None'
}

class MultiSelectDropdown extends React.Component {
  constructor(props) {
    super(props)

    this.state = {
      isOpen: false,
      localSelectedItems: props.selectedItems,
    }
  }

  handleClickOutside() {
    this.setState({isOpen: false})
  }

  componentWillReceiveProps(nextProps) {
    if (!_.isEqual(this.props.selectedItems, nextProps.selectedItems)) {
      return
    }

    this.setState({localSelectedItems: nextProps.selectedItems})
  }

  toggleMenu = e => {
    e.stopPropagation()
    this.setState({isOpen: !this.state.isOpen})
  }

  onSelect = (item, e) => {
    e.stopPropagation()

    const {onApply, isApplyShown} = this.props
    const {localSelectedItems} = this.state

    let nextItems
    if (this.isSelected(item)) {
      nextItems = localSelectedItems.filter(i => i.name !== item.name)
    } else {
      nextItems = [...localSelectedItems, item]
    }

    if (!isApplyShown) {
      onApply(nextItems)
    }

    this.setState({localSelectedItems: nextItems})
  }

  isSelected(item) {
    return !!this.state.localSelectedItems.find(({name}) => name === item.name)
  }

  handleApply = e => {
    e.stopPropagation()

    this.setState({isOpen: false})
    this.props.onApply(this.state.localSelectedItems)
  }

  render() {
    const {localSelectedItems, isOpen} = this.state
    const {label, buttonSize, buttonColor, customClass, iconName} = this.props

    return (
      <div className={classnames(`dropdown ${customClass}`, {open: isOpen})}>
        <div
          onClick={this.toggleMenu}
          className={`dropdown-toggle btn ${buttonSize} ${buttonColor}`}
        >
          {iconName ? <span className={`icon ${iconName}`} /> : null}
          <span className="dropdown-selected">
            {labelText({localSelectedItems, isOpen, label})}
          </span>
          <span className="caret" />
        </div>
        {this.renderMenu()}
      </div>
    )
  }

  renderMenu() {
    const {items, isApplyShown} = this.props
    const applyButton = isApplyShown
      ? <li className="multi-select--apply">
          <button className="btn btn-xs btn-info" onClick={this.handleApply}>
            Apply
          </button>
        </li>
      : null

    return (
      <ul className="dropdown-menu">
        {applyButton}
        <FancyScrollbar
          autoHide={false}
          autoHeight={true}
          maxHeight={DROPDOWN_MENU_MAX_HEIGHT}
        >
          {items.map((listItem, i) => {
            return (
              <li
                key={i}
                className={classnames('multi-select--item', {
                  checked: this.isSelected(listItem),
                })}
                onClick={_.wrap(listItem, this.onSelect)}
              >
                <div className="multi-select--checkbox" />
                <span>
                  {listItem.name}
                </span>
              </li>
            )
          })}
        </FancyScrollbar>
      </ul>
    )
  }
}

const {arrayOf, bool, func, shape, string} = PropTypes

MultiSelectDropdown.propTypes = {
  onApply: func.isRequired,
  items: arrayOf(
    shape({
      name: string.isRequired,
    })
  ).isRequired,
  selectedItems: arrayOf(
    shape({
      name: string.isRequired,
    })
  ),
  label: string,
  buttonSize: string,
  buttonColor: string,
  customClass: string,
  iconName: string,
  isApplyShown: bool,
}

MultiSelectDropdown.defaultProps = {
  buttonSize: 'btn-sm',
  buttonColor: 'btn-default',
  customClass: 'dropdown-160',
  selectedItems: [],
  isApplyShown: true,
}

export default onClickOutside(MultiSelectDropdown)
