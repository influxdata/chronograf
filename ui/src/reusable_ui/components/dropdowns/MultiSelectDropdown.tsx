// Libraries
import React, {Component, CSSProperties, Fragment} from 'react'
import classnames from 'classnames'
import _ from 'lodash'

// Components
import {ClickOutside} from 'src/shared/components/ClickOutside'
import DropdownDivider from 'src/reusable_ui/components/dropdowns/DropdownDivider'
import DropdownItem from 'src/reusable_ui/components/dropdowns/DropdownItem'
import DropdownButton from 'src/reusable_ui/components/dropdowns/DropdownButton'
import FancyScrollbar from 'src/shared/components/FancyScrollbar'

// Types
import {
  DropdownMenuColors,
  ComponentStatus,
  ComponentColor,
  ComponentSize,
  IconFont,
} from 'src/reusable_ui/types'

import {ErrorHandling} from 'src/shared/decorators/errors'

interface Props {
  children: JSX.Element[]
  onChange: (selectedIDs: string[], value: any) => void
  onCollapse?: () => void
  selectedIDs: string[]
  buttonColor?: ComponentColor
  buttonSize?: ComponentSize
  menuColor?: DropdownMenuColors
  status?: ComponentStatus
  widthPixels?: number
  icon?: IconFont
  wrapText?: boolean
  customClass?: string
  maxMenuHeight?: number
  emptyText?: string
  separatorText?: string
}

interface State {
  expanded: boolean
}

@ErrorHandling
class MultiSelectDropdown extends Component<Props, State> {
  public static defaultProps: Partial<Props> = {
    buttonColor: ComponentColor.Default,
    buttonSize: ComponentSize.Small,
    status: ComponentStatus.Default,
    wrapText: false,
    maxMenuHeight: 250,
    menuColor: DropdownMenuColors.Sapphire,
    emptyText: 'Choose an item',
    separatorText: ', ',
  }

  public static Button = DropdownButton
  public static Item = DropdownItem
  public static Divider = DropdownDivider

  constructor(props: Props) {
    super(props)

    this.state = {
      expanded: false,
    }
  }

  public render() {
    this.validateChildCount()

    return (
      <ClickOutside onClickOutside={this.collapseMenu}>
        <div className={this.containerClassName} style={this.containerStyle}>
          {this.button}
          {this.menuItems}
        </div>
      </ClickOutside>
    )
  }

  private toggleMenu = (): void => {
    this.setState({expanded: !this.state.expanded})
  }

  private collapseMenu = (): void => {
    const {onCollapse} = this.props

    this.setState({expanded: false})

    if (onCollapse) {
      onCollapse()
    }
  }

  private get containerStyle(): CSSProperties {
    const {widthPixels} = this.props

    if (widthPixels) {
      return {width: `${widthPixels}px`}
    }

    return {width: '100%'}
  }

  private get containerClassName(): string {
    const {buttonColor, buttonSize, status, wrapText, customClass} = this.props

    return classnames(
      `dropdown dropdown-${buttonSize} dropdown-${buttonColor}`,
      {
        disabled: status === ComponentStatus.Disabled,
        'dropdown-wrap': wrapText,
        [customClass]: customClass,
      }
    )
  }

  private get button(): JSX.Element {
    const {
      selectedIDs,
      status,
      buttonColor,
      buttonSize,
      icon,
      children,
      emptyText,
      separatorText,
    } = this.props
    const {expanded} = this.state

    const selectedChildren = children.filter(child =>
      _.includes(selectedIDs, child.props.id)
    )

    let label: string | Array<string | JSX.Element>

    if (selectedChildren.length) {
      label = selectedChildren.map((sc, i) => {
        if (i < selectedChildren.length - 1) {
          return (
            <Fragment key={sc.props.id}>
              {sc.props.children}
              {separatorText}
            </Fragment>
          )
        }

        return sc.props.children
      })
    } else {
      label = emptyText
    }

    return (
      <DropdownButton
        active={expanded}
        color={buttonColor}
        size={buttonSize}
        icon={icon}
        onClick={this.toggleMenu}
        status={status}
      >
        {label}
      </DropdownButton>
    )
  }

  private get menuItems(): JSX.Element {
    const {selectedIDs, maxMenuHeight, menuColor, children} = this.props
    const {expanded} = this.state

    if (expanded) {
      return (
        <div
          className={`dropdown--menu-container dropdown--${menuColor}`}
          style={this.menuStyle}
        >
          <FancyScrollbar
            autoHide={false}
            autoHeight={true}
            maxHeight={maxMenuHeight}
          >
            <div className="dropdown--menu" data-test="dropdown-menu">
              {React.Children.map(children, (child: JSX.Element) => {
                if (this.childTypeIsValid(child)) {
                  if (child.type === DropdownItem) {
                    return (
                      <DropdownItem
                        {...child.props}
                        key={child.props.id}
                        checkbox={true}
                        selected={_.includes(selectedIDs, child.props.id)}
                        onClick={this.handleItemClick}
                      >
                        {child.props.children}
                      </DropdownItem>
                    )
                  }

                  return (
                    <DropdownDivider {...child.props} key={child.props.id} />
                  )
                } else {
                  throw new Error(
                    'Expected children of type <Dropdown.Item /> or <Dropdown.Divider />'
                  )
                }
              })}
            </div>
          </FancyScrollbar>
        </div>
      )
    }

    return null
  }

  private get menuStyle(): CSSProperties {
    const {wrapText, widthPixels} = this.props

    let containerWidth = '100%'

    if (widthPixels) {
      containerWidth = `${widthPixels}px`
    }

    if (wrapText && widthPixels) {
      return {
        width: containerWidth,
      }
    }

    return {
      minWidth: containerWidth,
    }
  }

  private handleItemClick = (value: any): void => {
    const {onChange, selectedIDs} = this.props
    let updatedSelection

    if (_.includes(selectedIDs, value.id)) {
      updatedSelection = selectedIDs.filter(id => id !== value.id)
    } else {
      updatedSelection = [...selectedIDs, value.id]
    }

    onChange(updatedSelection, value)
  }

  private validateChildCount = (): void => {
    const {children} = this.props

    if (React.Children.count(children) === 0) {
      throw new Error(
        'Dropdowns require at least 1 child element. We recommend using Dropdown.Item and/or Dropdown.Divider.'
      )
    }
  }

  private childTypeIsValid = (child: JSX.Element): boolean =>
    child.type === DropdownItem || child.type === DropdownDivider
}

export default MultiSelectDropdown
