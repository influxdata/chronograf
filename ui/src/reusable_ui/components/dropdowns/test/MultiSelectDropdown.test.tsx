import React from 'react'
import {shallow} from 'enzyme'

import MultiSelectDropdown from 'src/reusable_ui/components/dropdowns/MultiSelectDropdown'
import DropdownButton from 'src/reusable_ui/components/dropdowns/DropdownButton'

describe('MultiSelectDropdown', () => {
  let wrapper

  const wrapperSetup = (override = {}) => {
    const items = [
      {
        id: 'item-a',
        text: 'A',
      },
      {
        id: 'item-b',
        text: 'B',
      },
      {
        id: 'item-c',
        text: 'C',
      },
    ]

    const selectedIDs = [items[0].id]

    const children = items.map(item => (
      <MultiSelectDropdown.Item id={item.id} key={item.id} value={item}>
        {item.text}
      </MultiSelectDropdown.Item>
    ))

    const props = {
      children,
      selectedIDs,
      onChange: () => {},
      ...override,
    }

    return shallow(<MultiSelectDropdown {...props} />)
  }

  beforeEach(() => {
    jest.resetAllMocks()
    wrapper = wrapperSetup()
  })

  it('mounts without exploding', () => {
    expect(wrapper).toHaveLength(1)
  })

  it('matches snapshot with minimal props', () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe('with menu expanded', () => {
    it('renders a dropdown menu', () => {
      const button = wrapper.find(DropdownButton)

      button.simulate('click')

      expect(wrapper.find('[data-test="dropdown-menu"]')).toHaveLength(1)
    })

    it('matches snapshot', () => {
      const button = wrapper.find(DropdownButton)

      button.simulate('click')

      expect(wrapper).toMatchSnapshot()
    })
  })
})
