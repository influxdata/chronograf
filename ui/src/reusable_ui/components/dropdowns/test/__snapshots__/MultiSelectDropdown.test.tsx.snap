// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultiSelectDropdown matches snapshot with minimal props 1`] = `
<ClickOutside
  onClickOutside={[Function]}
>
  <div
    className="dropdown dropdown-sm dropdown-default"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <DropdownButton
      active={false}
      color="default"
      onClick={[Function]}
      size="sm"
      status="default"
    >
      A
    </DropdownButton>
  </div>
</ClickOutside>
`;

exports[`MultiSelectDropdown with menu expanded matches snapshot 1`] = `
<ClickOutside
  onClickOutside={[Function]}
>
  <div
    className="dropdown dropdown-sm dropdown-default"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <DropdownButton
      active={true}
      color="default"
      onClick={[Function]}
      size="sm"
      status="default"
    >
      A
    </DropdownButton>
    <div
      className="dropdown--menu-container dropdown--sapphire"
      style={
        Object {
          "minWidth": "100%",
        }
      }
    >
      <FancyScrollbar
        autoHeight={true}
        autoHide={false}
        maxHeight={250}
        setScrollTop={[Function]}
        style={Object {}}
      >
        <div
          className="dropdown--menu"
          data-test="dropdown-menu"
        >
          <DropdownItem
            checkbox={true}
            id="item-a"
            key=".$item-a"
            onClick={[Function]}
            selected={true}
            value={
              Object {
                "id": "item-a",
                "text": "A",
              }
            }
          >
            A
          </DropdownItem>
          <DropdownItem
            checkbox={true}
            id="item-b"
            key=".$item-b"
            onClick={[Function]}
            selected={false}
            value={
              Object {
                "id": "item-b",
                "text": "B",
              }
            }
          >
            B
          </DropdownItem>
          <DropdownItem
            checkbox={true}
            id="item-c"
            key=".$item-c"
            onClick={[Function]}
            selected={false}
            value={
              Object {
                "id": "item-c",
                "text": "C",
              }
            }
          >
            C
          </DropdownItem>
        </div>
      </FancyScrollbar>
    </div>
  </div>
</ClickOutside>
`;
